src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,78 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { addComment } from '../api/ApiMethods';
+
+type Props = {
+  openPost: Post | null,
+  setPostComments: React.Dispatch<React.SetStateAction<Comment[]>>,
+  setHasError: (v: boolean) => void,
+  setOpenForm: (v: boolean) => void,
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  openPost,
+  setPostComments,
+  setHasError,
+  setOpenForm,
+}) => {
+  const [formInputs, setFormInputs] = useState({
+    name: '',
+    email: '',
+    body: '',
+  });
+  const [loading, setLoading] = useState(false);
+  const [tuched, setTuched] = useState(false);
+
+  const checkInputName = !formInputs.name.trim() && tuched;
+  const checkEmailName = !formInputs.email.trim() && tuched;
+  const checkBodyName = !formInputs.body.trim() && tuched;
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement
+  | HTMLTextAreaElement>) => {
+    const { name, value } = e.target;
+
+    setTuched(false);
+    setFormInputs(prev => ({
+      ...prev,
+      [name]: value,
+    }));
+  };
+
+  const handleClear = () => {
```

Comment:
you should also clear errors (one of tests fails because of this):
![image](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/86923089/c11b272e-1709-42df-9b4c-78f24c4bc081)